src/P3Scheme.jl

```diff
@@ -667,4 +767,464 @@ function D_m(p3::PSP3, q::FT, N::FT, ρ_r::FT, F_r::FT) where {FT}
+    q_c::FT,
+    N_c::FT,
+    ρ_a::FT,
+    F_r::FT,
+    ρ_r::FT,
+    T::FT,
+    E_ci = FT(1),
+) where {FT}
+    ρ_w = FT(1000)      # TO DO: add this as density_liquid_water from ClimaParams
+    th = thresholds(p3, ρ_r, F_r)
+    (λ, N_0) = distribution_parameter_solver(p3, qᵢ, Nᵢ, ρ_r, F_r)
+    (colliding_bound, ice_bound) =
+        integration_bounds(type, p3, eps(FT), λ, N_0, Nᵢ, q_c, N_c)
+
+    if T > 273.15
+        f_warm(D_c, Dᵢ) =
+            E_ci / ρ_a *
+            N′colliding(type, D_c, q_c, N_c) *
+            N′ice(Dᵢ, p3, λ, N_0) *
+            a(D_c, Dᵢ) *
+            p3_mass(p3, Dᵢ, F_r, th) *
+            vel_diff(type, D_c, Dᵢ, p3, Chen2022, ρ_a, F_r, th)
+        (dqdt, error) = HC.hcubature(
+            d -> f_warm(d[1], d[2]),
+            (FT(0), FT(0)),
+            (colliding_bound, ice_bound),
+        )
+    else
+        f_cold(D_c, Dᵢ) =
+            E_ci / ρ_a *
+            N′colliding(type, D_c, q_c, N_c) *
+            N′ice(Dᵢ, p3, λ, N_0) *
+            a(D_c, Dᵢ) *
+            mass_s(D_c, ρ_w) *
+            vel_diff(type, D_c, Dᵢ, p3, Chen2022, ρ_a, F_r, th)
+        (dqdt, error) = HC.hcubature(
+            d -> f_cold(d[1], d[2]),
+            (FT(0), FT(0)),
+            (colliding_bound, ice_bound),
+        )
+    end
+
+    return dqdt
+end
+
+"""
+    dmdt_mass(p3, D, F_r, th) 
+
+ - p3 - a struct containing p3 parameters 
+ - D - maximum dimension of the particle 
+ - F_r - rime mass fraction (q_rim/ q_i) 
+ - th - thresholds as calculated by thresholds()
+
+Returns the value equivalent to dm(D)/dt * 4 / D for each P3 regime
+    4 / D comes from dD/dt
+"""
+function dmdt_mass(p3, D, F_r, th)
+    D_th = D_th_helper(p3)
+    if D_th > D
+        return 2 * π * p3.ρ_i * D
+    elseif F_r == 0
+        return 4 * α_va_si(p3) * p3.β_va * D^(p3.β_va - 2)
+    elseif th.D_gr > D >= D_th
+        return 4 * α_va_si(p3) * p3.β_va * D^(p3.β_va - 2)
+    elseif th.D_cr > D >= th.D_gr
+        return 2 * π * th.ρ_g * D
+    elseif D >= th.D_cr
+        return 4 * α_va_si(p3) / (1 - F_r) * p3.β_va * D^(p3.β_va - 2)
+    end
+end
+
+"""
+    p3_melt(p3, Chen2022, aps, tps, q, N, ρ, T, ρ_a, F_r, ρ_r)
+
+ - p3 - a struct containing p3 parameters 
+ - Chen2022 - struct containing Chen 2022 velocity parameters 
+ - aps - air properties
+ - tps - thermodynamics parameters
+ - q - mass mixing ratio of ice 
+ - N - number mixing ratio of ice 
+ - T - temperature (K) 
+ - ρ_a - air density
+ - F_r - rime mass fraction (q_rim/ q_i)
+ - ρ_r - rime density (q_rim/B_rim) 
+
+ Returns the calculated melting rate of ice
+ Equivalent to the measure of QIMLT in Morrison and Mildbrandt (2015) 
+"""
+function p3_melt(
+    p3::PSP3,
+    Chen2022::CMP.Chen2022VelType,
+    aps::CMP.AirProperties{FT},
+    tps::TDP.ThermodynamicsParameters{FT},
+    q::FT,
+    N::FT,
+    T::FT,
+    ρ_a::FT,
+    F_r::FT,
+    ρ_r::FT,
+) where {FT}
+    # Get constants
+    (; ν_air, D_vapor, K_therm) = aps
+    a = FT(0.78)
```

Comment:
Could you remind me wehre are this values coming from? Should we also add them to the `p3` parameters struct?

Reply:
These values are coming from the A. Seifert and K. D. Beheng. paper that is cited in the 1M scheme ventilation factor section. The P3 Scheme only says that they use an approach similar to a paper which just gives an equation that we are not using due to the differentiation in snow, graupel, and ice within that paper. I am not sure whether these are the values we want to use but the P3 paper does not give an answer to this question.